java/src/com/google/idea/blaze/java/sync/source/JavaSourcePackageReader.java

```diff
@@ -62,11 +65,19 @@ public String getDeclaredPackageOfJavaFile(
     try (InputStream javaInputStream = inputStreamProvider.forFile(sourceFile)) {
       BufferedReader javaReader = new BufferedReader(new InputStreamReader(javaInputStream, UTF_8));
       String javaLine;
-
+      boolean isScala = sourceFile.getName().endsWith(".scala");
+      List<String> packageDeclarations = Lists.newArrayList();
       while ((javaLine = javaReader.readLine()) != null) {
         Matcher packageMatch = PACKAGE_PATTERN.matcher(javaLine);
         if (packageMatch.find()) {
-          return packageMatch.group(1);
```

Comment:
How about breaking the loop here in case it is not a scala file? In Java typically it's the first line, so there's no point of iterating through all the files